pkg/proxy/iptables/proxier.go

```diff
@@ -294,10 +294,14 @@ func NewProxier(ipt utiliptables.Interface,
 		ipFamily = v1.IPv6Protocol
 	}
 
-	var incorrectAddresses []string
-	nodePortAddresses, incorrectAddresses = utilproxy.FilterIncorrectCIDRVersion(nodePortAddresses, ipFamily)
-	if len(incorrectAddresses) > 0 {
-		klog.Warningf("NodePortAddresses of wrong family; %s", incorrectAddresses)
+	isCIDR := true
+	ipFamilyIPMap := utilproxy.MapIPsToIPFamily(nodePortAddresses, isCIDR)
+	nodePortAddresses = ipFamilyIPMap[ipFamily]
+	for ipFam, ips := range ipFamilyIPMap {
```

Comment:
You really don't need to loop here. Just define another key for altFamily and use it.

Reply:
In fact, why not `util.OtherIPFamily(x)` which returns whichever family is not x ?  `ipFamilyIPMap[OtherIPFamily(ipFamily)]`

Reply:
@khenidak @thockin Got it. I'll add OtherIPFamily in utils and use it instead of looping over the map

Reply:
Implemented utilproxy.OtherIPFamily(x) to get the other IP family and this avoids looping over the map